### PR TITLE
honk: 0.9.91 -> 1.0.0

### PR DESCRIPTION
**Pull request by drupol**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/honk/default.nix [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
   subPackages = [ "." ];
 
   postPatch = ''
-    substituteInPlace honk.go --replace \
+    substituteInPlace main.go --replace \
       "var viewDir = \".\"" \
```
Comment (doronbehar) [inline pkgs/servers/honk/default.nix:30]: Ideally upstream would have supported changing this variable via an ldFlag or a command line argument...

**Review by doronbehar — APPROVED** [review]
LGTM, haven't tested.

File: pkgs/servers/honk/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/servers/honk/default.nix:30]: That's actually possible to add the `-viewdir` parameter to honk. This is just to have something working out of the box without adding any extra argument.

Comment (doronbehar) [reply pkgs/servers/honk/default.nix:30]: > That's actually possible to add the `-viewdir` parameter to honk. This is just to have something working out of the box without adding any extra argument.

I see. I saw your [honk module PR](https://github.com/NixOS/nixpkgs/pull/249765) and I won't be able to review it. But I think it's OK to keep this substitution as is, but perhaps it'd be ideal to maybe remove it in favor of the module taking care of it. I am just thinking out loud, no action is strictly requested :).

Comment (doronbehar) [reply pkgs/servers/honk/default.nix:30]: Of course the option of having this default parameter set in `ldFlags` is even better.

Comment (drupol) [reply pkgs/servers/honk/default.nix:30]: I agree. Is it ok for you if I merge this?

Comment (doronbehar) [reply pkgs/servers/honk/default.nix:30]: Maybe it'd be nice to add a comment near that hook that ideally the situation could be better, or that this is just for setting the default value of this parameter, something like that.

Comment (drupol) [reply pkgs/servers/honk/default.nix:30]: I will immediately do that.

Comment (drupol) [reply pkgs/servers/honk/default.nix:33]: Done, is it ok?

Comment (doronbehar) [reply pkgs/servers/honk/default.nix:33]: Yes good enough :) LGTM now, when CI is green.

**Pull request merged**